iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -222,6 +223,7 @@ public void commitDropTable(org.apache.hadoop.hive.metastore.api.Table hmsTable,
   }
 
   @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
Could we decrease the complexity instead of suppressing?

Reply:
Extracted the replace columns logic to a new method to decrease the complexity of the preAlterTable method.